src/python_testing/post_certification_tests/post-cert-checks.py

```diff
@@ -0,0 +1,348 @@
+    import fetch_paa_certs_from_dcl
+
+
+@dataclass
+class Failure:
+    test: str
+    step: str
+
+
+class Hooks():
+    def __init__(self):
+        self.failures = []
+        self.current_step = 'unknown'
+        self.current_test = 'unknown'
+
+    def start(self, count: int):
+        pass
+
+    def stop(self, duration: int):
+        pass
+
+    def test_start(self, filename: str, name: str, count: int):
+        self.current_test = name
+        pass
+
+    def test_stop(self, exception: Exception, duration: int):
+        pass
+
+    def step_skipped(self, name: str, expression: str):
+        pass
+
+    def step_start(self, name: str):
+        self.current_step = name
+
+    def step_success(self, logger, logs, duration: int, request):
+        pass
+
+    def step_failure(self, logger, logs, duration: int, request, received):
+        self.failures.append(Failure(self.current_test, self.current_step))
+
+    def step_unknown(self):
+        pass
+
+    def get_failures(self) -> list[str]:
+        return self.failures
+
+
+class TestEventTriggersCheck(MatterBaseTest, BasicCompositionTests):
+    @async_test_body
+    async def test_TestEventTriggersCheck(self):
+        self.connect_over_pase(self.default_controller)
+        gd = Clusters.GeneralDiagnostics
+        ret = await self.read_single_attribute_check_success(cluster=gd, attribute=gd.Attributes.TestEventTriggersEnabled)
+        asserts.assert_equal(ret, 0, "TestEventTriggers are still on")
+
+
+def get_dcl_vendor(vid):
+    return requests.get(f"{fetch_paa_certs_from_dcl.PRODUCTION_NODE_URL_REST}/dcl/vendorinfo/vendors/{vid}").json()
+
+
+def get_dcl_model(vid, pid):
+    return requests.get(f"{fetch_paa_certs_from_dcl.PRODUCTION_NODE_URL_REST}/dcl/model/models/{vid}/{pid}").json()
+
+
+def get_dcl_compliance_info(vid, pid, software_version):
+    return requests.get(f"{fetch_paa_certs_from_dcl.PRODUCTION_NODE_URL_REST}/dcl/compliance/compliance-info/{vid}/{pid}/{software_version}/matter").json()
+
+
+def get_dcl_certified_model(vid, pid, software_version):
+    return requests.get(f"{fetch_paa_certs_from_dcl.PRODUCTION_NODE_URL_REST}/dcl/compliance/certified-models/{vid}/{pid}/{software_version}/matter").json()
+
+
+class DclCheck(MatterBaseTest, BasicCompositionTests):
+    @async_test_body
+    async def setup_class(self):
+        self.connect_over_pase(self.default_controller)
+        bi = Clusters.BasicInformation
+        self.vid = await self.read_single_attribute_check_success(cluster=bi, attribute=bi.Attributes.VendorID)
+        self.pid = await self.read_single_attribute_check_success(cluster=bi, attribute=bi.Attributes.ProductID)
+        self.software_version = await self.read_single_attribute_check_success(cluster=bi, attribute=bi.Attributes.SoftwareVersion)
+
+    def steps_Vendor(self):
+        return [TestStep(1, "Check if device VID is listed in the DCL vendor schema", "Listing found")]
+
+    def test_Vendor(self):
```

Comment:
The following should be optional or good to show as informative to be missing since they are not mandatory to be public/available at Certification Testing stage. (specially the Compliance-related ones, as those require the cert tests to be completed in order to get).
- test_Vendor
- test_Model  <- Missing "test_ModelVersion"
- test_Compliance
- test_CertifiedModel

The only one that could make sense to confirm as mandatory is the existence of the Root CA in the DCL (to define if Main-Net or Test-Net).

Reply:
So this test is going to be run after cert on production devices at the interop lab. For production devices, these are all mandatory, right?

what's test_Model/Version?

Reply:
Still not mandatory, a company may days or even months to decide going public in the DCL after they get their certification approved... But I do believe it's good to have for members to be aware of what they are missing.

In DCL, there is Model and Model-Version (that contains the Software Version) under the Model. 
I believe the test is only checking for the Model, not the Model-Version... which is the entry that actually gets the Compliance entry associated with it.

When a Product gets a certification, it is granted to an specific software version (VID/PID/SoftwareVersion/SoftwareVersionString) that translates to a Model-Version in the DCL. Sometimes members just enter the Model to the DCL, and they miss to enter the Model-Version, so they won't get their Compliance entry in the DCL until they add it. Or sometimes the information they add in the Model-Version mismatches with the information in the Certification Tool, for example:
Software Version String in Cert Tool: "v1.0.0"
Software Version String in DCL: "1.0" or "1.0.0"

That kind of discrepancy triggers troubleshooting with the Certification team to make sure all data is aligned. This could be caught at early stage.

Some info: [HowTo - Writing to the DCL.pdf](https://groups.csa-iot.org/wg/members-all/document/27881)

Reply:
Yes, sorry I'm not being clear. 
Your script is validating that a Model is in the DCL, which is fine. But it still requires checking that the corresponding Model-Version is also in the DCL. That is the actual product in the DCL, not the Model.
So, the script will confirm that the data from the Basic Information Cluster matches the information from the DCL, and that is with the Model (contains the VID, PID and Device Type) and Model-Version (contains the VID, PID, Software Version and Software Version String), not just the Model.

Reply:
Is the model version not required to be in the DCL once the product is released? This test is meant to be run on launched products in the interop lab - ie things the lab purchased from a store. At that point, there's no confidentiality problem because it's a commercial product, and the commissioners should be able to get information about the device version from the production DCL. Or is the model version stuff completely optional (ie, ecosystems shouldn't be checking that).

Reply:
spoke offline - these checks are correct for devices in production. Renamed script and updated the top of file readme to reflect that.